controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java

```diff
@@ -188,7 +188,10 @@ public Change withCompatibilityPlatform(Change change, InstanceName instance) {
                     .stream() // Pick the latest platform with appropriate confidence, which is compatible with the compile version.
                     .filter(compatibleWithCompileVersion)
                     .findFirst()
-                    .map(platform -> change.withoutPin().with(platform))
+                    .map(platform -> change.platform()
+                                           .filter(platform::equals)
```

Comment:
Hmm, perhaps this should instead be a check on whether that platform is compatible with the compile version. That could also be done outside all of this ... Let me fix.